### PR TITLE
Add server-side merge of assignment changes

### PR DESCRIPTION
**Pull request by AltGr**

### CODE REVIEW

File: src/state/learnocaml_data.ml [hunk 1]
```diff
@@ -444,6 +444,52 @@ module Exercise = struct
       in
       by_status_explicit base a |> STM.bindings
 
+    let three_way_merge ~ancestor ~theirs ~ours =
```
Comment (yurug) [inline src/state/learnocaml_data.ml]: I am not sure to understand this function: can the merge encounter conflicts? Can it fail?

File: src/state/learnocaml_data.mli [hunk 2]
```diff
@@ -178,6 +178,12 @@ module Exercise: sig
     val by_status:
       Token.Set.t -> assignments -> (status * Token.Set.t) list
 
+    (* Merges all changes from [theirs] and [ours], based on [ancestor]. [ours]
```
Comment (yurug) [inline src/state/learnocaml_data.mli]: By privileging "ours", you choose a last-arrived-get-the-prize strategy right?

Comment (AltGr) [reply src/state/learnocaml_data.mli]: yes, as mentionned in the commit notes. This happens for conflicting changes on same exercise + same student.
If we choose the alternative (refuse conflicting changes), we need to add a way to report the partial error to the client.

Comment (yurug) [reply src/state/learnocaml_data.mli]: That's the good design choice I think. I just wanted to check I correctly understood.

**Pull request merged**